scripts/core/editor3/reducers/suggestions.jsx

```diff
@@ -133,13 +135,37 @@ const pasteAddSuggestion = (state, {content, data}) => {
  * @return {Object} returns new state
  * @description Accept or reject the suggestions in the selection.
  */
-const processSuggestion = (state, {selection}, accepted) => {
+const processSuggestion = (state, {suggestion}, accepted) => {
+    const {selection} = suggestion;
     const start = selection.getStartOffset();
     const end = selection.getEndOffset();
     let {editorState} = state;
     let style;
     let data;
 
+    const resolvedSuggestions = getCustomDataFromEditor(
```

Comment:
You can extract the code that deals with suggestion history in a separate function or even in a separate file.